Reverts:
------------------------------------------------------------
revno: 335 [merge]
author: Charles Kerr <charles.kerr@canonical.com>
committer: Tarmac
branch nick: trunk.13.04
timestamp: Wed 2013-01-23 08:29:34 +0000
message:
  remove g_type_init() calls, bump glib requirement to 2.35.4. Fixes: https://bugs.launchpad.net/bugs/1103087.
  
  Approved by Lars Uebernickel, PS Jenkins bot.
------------------------------------------------------------
=== modified file 'configure.ac'
--- configure.ac	2013-01-22 20:30:05 +0000
+++ configure.ac	2012-11-08 15:39:07 +0000
@@ -41,7 +41,7 @@
 GTK_REQUIRED_VERSION=3.5.18
 GIO_UNIX_REQUIRED_VERSION=2.33.10
 INDICATOR_REQUIRED_VERSION=0.3.19
-GLIB_REQUIRED_VERSION=2.35.4
+GLIB_REQUIRED_VERSION=2.33.10
 INTROSPECTION_REQUIRED_VERSION=1.32.0
 
 PKG_CHECK_MODULES(APPLET, gtk+-3.0 >= $GTK_REQUIRED_VERSION

=== modified file 'doc/reference/Makefile.am'
--- doc/reference/Makefile.am	2013-01-22 20:23:57 +0000
+++ doc/reference/Makefile.am	2012-08-29 11:27:54 +0000
@@ -4,6 +4,7 @@
 
 DOC_SOURCE_DIR = $(top_srcdir)/libmessaging-menu
 
+SCANGOBJ_OPTIONS=--type-init-func="g_type_init()"
 MKDB_OPTIONS=--xml-mode --output-format=xml
 
 # Used for dependencies. The docs will be rebuilt if any of these change.

=== modified file 'src/messages-service.c'
--- src/messages-service.c	2013-01-22 20:23:57 +0000
+++ src/messages-service.c	2012-10-02 21:19:53 +0000
@@ -620,8 +620,11 @@
 int
 main (int argc, char ** argv)
 {
-	GMainLoop * mainloop;
-	IndicatorService * service;
+	GMainLoop * mainloop = NULL;
+	IndicatorService * service = NULL;
+
+	/* Glib init */
+	g_type_init();
 
 	gdk_init(&argc, &argv);
 	mainloop = g_main_loop_new (NULL, FALSE);

=== modified file 'test/indicator-messages-service-activate.c'
--- test/indicator-messages-service-activate.c	2013-01-22 20:23:57 +0000
+++ test/indicator-messages-service-activate.c	2009-05-28 15:29:41 +0000
@@ -28,6 +28,8 @@
 int
 main (int argc, char ** argv)
 {
+	g_type_init();
+
 	guint returnval = 0;
 	GError * error = NULL;
 

=== modified file 'test/test-gactionmuxer.cpp'
--- test/test-gactionmuxer.cpp	2013-01-22 20:23:57 +0000
+++ test/test-gactionmuxer.cpp	2012-06-04 21:47:13 +0000
@@ -25,6 +25,8 @@
 TEST(GActionMuxerTest, Sanity) {
 	GActionMuxer *muxer;
 
+	g_type_init ();
+
 	g_action_muxer_insert (NULL, NULL, NULL);
 	g_action_muxer_remove (NULL, NULL);
 
@@ -44,6 +46,8 @@
 	GActionMuxer *muxer;
 	gchar **actions;
 
+	g_type_init ();
+
 	muxer = g_action_muxer_new ();
 
 	actions = g_action_group_list_actions (G_ACTION_GROUP (muxer));
@@ -63,6 +67,8 @@
 	GActionMuxer *muxer;
 	gchar **actions;
 
+	g_type_init ();
+
 	group1 = g_simple_action_group_new ();
 	g_simple_action_group_add_entries (group1,
 					   entries1,
@@ -154,6 +160,8 @@
 	GVariant *state_hint[2];
 	GVariant *state[2];
 
+	g_type_init ();
+
 	group = g_simple_action_group_new ();
 	action = g_simple_action_new ("one", G_VARIANT_TYPE_STRING);
 	g_simple_action_group_insert (group, G_ACTION (action));

